### PR TITLE
Specified Exception to ApiException,IOException

### PR DESCRIPTION
**Pull request by markmcd**
PTAL @broady @domesticmouse

### CODE REVIEW

File: src/main/java/com/google/maps/internal/OkHttpPendingResult.java [hunk 1]
```diff
@@ -147,11 +148,16 @@ public void onResponse(Response response) throws IOException {
...
+    try {
+      QueuedResponse r = waiter.take();
+      if (r.response != null) {
+        return parseResponse(r.request, r.response);
+      } else {
+        throw r.e;
+      }
+    } catch (InterruptedException e) {
+      LOG.log(Level.SEVERE, "Thread interrupted while executing", e);
+      return null;
```
Comment (broady) [inline src/main/java/com/google/maps/internal/OkHttpPendingResult.java]: Is `null` correct here? I feel like we should throw *something*.

File: src/main/java/com/google/maps/internal/OkHttpPendingResult.java [hunk 2]
```diff
@@ -109,7 +110,7 @@ public QueuedResponse(OkHttpPendingResult<T, R> request, Exception e) {
   }
 
   @Override
-  public T await() throws Exception {
+  public T await() throws ApiException, IOException {
```
Comment (broady) [inline src/main/java/com/google/maps/internal/OkHttpPendingResult.java]: As discussed, we should probably catch `IOException`s and retry.

**Comment by domesticmouse** [comment]
@markmcd Should we fix this PR up, or should I just close it out?

**Comment by domesticmouse** [comment]
Closing this as it is out of date.

**Pull request closed without merging**